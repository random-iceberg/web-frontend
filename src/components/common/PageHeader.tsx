/*
Draft Doc

This Component is for Page Headers, Use this to keep page header layout uniform across the app.
*/
import React from "react";

interface PageHeaderProps {
  title: string;
  description: string;
}

const PageHeader: React.FC<PageHeaderProps> = ({ title, description }) => {
  return (
    <div className="mb-8">
      <h1 className="text-3xl font-bold text-gray-900 mb-2">{title}</h1>
      <p className="text-lg text-gray-600">{description}</p>
    </div>
  );
};

export default PageHeader;
