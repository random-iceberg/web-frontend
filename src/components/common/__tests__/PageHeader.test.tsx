import React from "react";
import { render } from "@testing-library/react";
import PageHeader from "../PageHeader";

describe("PageHeader Component", () => {
  it("renders correctly with title and description", () => {
    const { asFragment } = render(
      <PageHeader
        title="Page Title"
        description="This is a description of the page."
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with different title and description", () => {
    const { asFragment } = render(
      <PageHeader
        title="Another Page"
        description="Another description here."
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
