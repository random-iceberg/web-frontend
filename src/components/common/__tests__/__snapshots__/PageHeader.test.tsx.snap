// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Component renders correctly with different title and description 1`] = `
<DocumentFragment>
  <div
    class="mb-8"
  >
    <h1
      class="text-3xl font-bold text-gray-900 mb-2"
    >
      Another Page
    </h1>
    <p
      class="text-lg text-gray-600"
    >
      Another description here.
    </p>
  </div>
</DocumentFragment>
`;

exports[`PageHeader Component renders correctly with title and description 1`] = `
<DocumentFragment>
  <div
    class="mb-8"
  >
    <h1
      class="text-3xl font-bold text-gray-900 mb-2"
    >
      Page Title
    </h1>
    <p
      class="text-lg text-gray-600"
    >
      This is a description of the page.
    </p>
  </div>
</DocumentFragment>
`;
